### PR TITLE
Moving special case of keff calculation (n<=3) to cpp side

### PR DESCRIPTION
**Pull request by matiaslavista**
Using test_data: 
https://anl.box.com/s/u1g3n8iai0u1n5f6ev3pg2j3ff941bqa

From Commit Below:
Change in Calculating K-Eff from cpp side for cases of batches (n) <= 3 has changed. Therefore filter_distribcell, which ran batches of 1,1,3 and 1, has been updated to reflect the more accurate calcultion. sourcepoint_batches explanation: This test needed to be updated because statepoint files are written for 3 or fewer batches, which will change the k-eff result written in write_eigenvalue_hdf5 for those statepoint files. All other k-effective values are unaffected and the final result of the simulation is the same.

### CODE REVIEW

File: src/eigenvalue.cpp [hunk 1]
```diff
@@ -388,10 +389,15 @@ int openmc_get_keff(double* k_combined)
   k_combined[0] = 0.0;
   k_combined[1] = 0.0;
 
-  // Make sure we have at least four realizations. Notice that at the end,
-  // there is a N-3 term in a denominator.
+  //Special case for n <=3. Notice that at the end,
+  //there is a N-3 term in a denominator.
```
Comment (smharper) [inline src/eigenvalue.cpp]: Add a space after the `//` for these comments.

File: src/eigenvalue.cpp [hunk 2]
```diff
@@ -388,10 +389,15 @@ int openmc_get_keff(double* k_combined)
...
   if (simulation::n_realizations <= 3) {
-    return -1;
+    k_combined[0] = simulation::keff;
+    k_combined[1] = simulation::keff_std;
+    if (simulation::n_realizations <=1){
```
Comment (smharper) [inline src/eigenvalue.cpp]: Add a space before the `{`

**Comment by paulromano** [comment]
Looks good to me too, other than the style issue already identified by @smharper. Also, congrats to @matiaslavista on his first PR!

File: src/eigenvalue.cpp (hunk 1 above)
Comment (matiaslavista) [reply src/eigenvalue.cpp]: Done

File: src/eigenvalue.cpp (hunk 2 above)
Comment (matiaslavista) [reply src/eigenvalue.cpp]: Done

**Comment by smharper** [comment]
Merged.  Thanks again, @matiaslavista!

**Pull request merged**